app/stac_api/models.py

```diff
@@ -125,48 +145,137 @@ class Collection(models.Model):
+        For all the given parameters this function checks, if the corresponding
+        parameters of the collection need to be updated. If so, they will be either
+        updated or an error will be raised, if updating fails.
+        '''
+        try:
+            if self.summaries["geoadmin:variant"] is None:
+
+                self.summaries["geoadmin:variant"] = [asset_geoadmin_variant]
+                self.save()
+
+            elif asset_geoadmin_variant not in self.summaries["geoadmin:variant"]:
+                self.summaries["geoadmin:variant"].append(asset_geoadmin_variant)
+                self.save()
 
+            if self.summaries["proj:epsg"] is None:
+                self.summaries["proj:epsg"] = [asset_proj_epsg]
+                self.save()
+
+            elif asset_proj_epsg not in self.summaries["proj:epsg"]:
+                self.summaries["proj:epsg"].append(asset_proj_epsg)
+                self.save()
+
+            if self.summaries["eo:gsd"] is None:
+                self.summaries["eo:gsd"] = [asset_eo_gsd]
+                self.save()
+
+            elif not float_in(asset_eo_gsd, self.summaries["eo:gsd"]):
+                self.summaries["eo:gsd"].append(asset_eo_gsd)
+                self.save()
+
+        except (KeyError, IndexError) as err:
+            logger.error(_(
+                "Error when updating collection's summaries values due to asset update: %s"), err
+                        )
+            raise ValidationError(_(
+                "Error when updating collection's summaries values due to asset update."
+            ))
+
+    def update_extent(self, item_properties_datetime):
+        '''
+        updates the collection's temporal extent when item's are update.
+        :param item_properties_datetime: item's value for properties_datetime.
+        This function checks, if the corresponding parameter of the collection
+        needs to be updated. If so, it will be either updated or an error will
+        be raised, if updating fails.
+        '''
+        try:
+
+            if self.extent["temporal"]["interval"][0][0] is None:
+
+                self.extent["temporal"]["interval"][0][0] = item_properties_datetime
+                self.save()
+
+            elif item_properties_datetime < self.extent["temporal"]["interval"][0][0]:
+
+                self.extent["temporal"]["interval"][0][0] = item_properties_datetime
+                self.save()
+
+            elif self.extent["temporal"]["interval"][0][1] is None:
+
+                self.extent["temporal"]["interval"][0][1] = item_properties_datetime
+                self.save()
+
+            elif item_properties_datetime > self.extent["temporal"]["interval"][0][1]:
+
+                self.extent["temporal"]["interval"][0][1] = item_properties_datetime
+                self.save()
+
+        except (KeyError, IndexError) as err:
+
+            logger.error(_('Updating the collection extent due to item update failed: %s'), err)
```

Comment:
I would not use `_()` getext in logging. Getext is for localization and we don't want to localize logging. All logging must be written in English. Localization should be kept for user interface message (e.g. like in the `ValidationError` where the message is for the user)